src/pages/index.js

```diff
@@ -27,44 +31,165 @@ function openPreviewModal(data) {
   const card = new Card(
     cardData,
     selectors.cardTemplate,
-    openPreviewModal
+    openPreviewModal,
+    handleCardDeleteClick,
+    handleLikeClick
   ).getCardElement();
   cardSection.addItem(card);
 }
 
-const userInfo = new UserInfo(".profile__title", ".profile__description");
+const userInfo = new UserInfo(".profile__title", ".profile__description", avatarImage);
 const profileEditForm = new PopupWithForm(
   "#profile-edit-modal",
   handleProfileEditSubmit
 );
 const cardEditForm = new PopupWithForm("#add-card-modal", handleCardFormSubmit);
 const previewImagePopup = new PopupWithImage("#modal__preview-card");
+const confirmDeletePopup = new PopupWithConfirm("#delete-card-modal");
+
+const avatarImagePopup = new PopupWithForm(
+  "#edit-avatar-modal",
+  handleAvatarFormSubmit
+);
 
 /*Validation*/
 
 const editFormValidator = new FormValidator(config, profileForm);
 const addCardValidator = new FormValidator(config, cardForm);
+const deleteCardValidator = new FormValidator(config, deleteCardForm);
+const avatarValidator = new FormValidator(config, avatarUpdateForm);
 
 /*Event Handlers*/
 
 function handleProfileEditSubmit(inputItems) {
-  userInfo.setUserInfo(inputItems);
-  profileEditForm.close();
+  profileEditForm.renderLoading(true);
+  api
+    .userProfileInfo(inputItems.name, inputItems.description)
+    .then((res) => {
+      console.log(res);
+      userInfo.setUserInfo(res);
+      profileEditForm.close();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      profileEditForm.renderLoading(false);
+    });
 }
-
 function handleCardFormSubmit(data) {
-  renderCard({ name: data["image-title"], link: data["image-url"] });
-  cardEditForm.close();
-  addCardValidator.resetValidation();
+  cardEditForm.renderLoading(true);
+  api
+    .addNewCard(data.name, data.link)
+    .then((res) => {
+      renderCard(res);
+      cardEditForm.close();
+      addCardValidator.resetValidation();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      cardEditForm.renderLoading(false);
+    });
+}
+
+function handleLikeClick(card) {
+  console.log(card);
+  if (card.isLiked) {
+    api
+      .dislikeCard(card._id)
+      .then(() => {
+        card.handleLikeButton();
+        card.isLiked = false;
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+  if (!card.isLiked) {
+    api
+      .likeCard(card._id)
+      .then(() => {
+        card.handleLikeButton();
+        card.isLiked = true;
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+}
+
+function handleCardDeleteClick(card) {
+  console.log(card);
+  confirmDeletePopup.open();
+  confirmDeletePopup.handleDelete(() => {
+    api
+      .deleteCard(card._id)
+      .then(() => {
+        confirmDeletePopup.close();
+        card.handleTrashButton();
+      })
+      .catch((err) => {
+        console.log(err);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors